js/controller.js

```diff
@@ -2452,6 +2462,14 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
 
       // validate language is available before update and save
       if (language && availableLanguages && _.contains(availableLanguages.languages, language)) {
+        // The act of changing the CC language doesn't currently enable captions
+        // automatically. The core will set the forceEnabled parameter to true when
+        // it is necessary to also enable captions themselves
+        if (params.forceEnabled) {
+          this.state.closedCaptionOptions.enabled = true;
+          this.state.persistentSettings.closedCaptionOptions.enabled = true;
```

Comment:
Is it possible to have a single source of truth for this (and possibly store persistent settings as an aside)?

Reply:
Yes but I think this is outside the scope of this PR. The closed captions logic in the skin also needs a good refactor, but it's a significant amount of effort.

Reply:
Sounds good